test/test_db.py

```diff
@@ -39,40 +46,66 @@ def teardown_function(function):
 
 
 def test_get_nick_id(db):
-    session = db.ssession()
-    tests = [
-        [None, 'embolalia', Identifier('Embolalia')],
-        # Ensures case conversion is handled properly
-        [None, '{}{}', Identifier('[]{}')],
-        # Unicode, just in case
-        [None, 'embölaliå', Identifier('EmbölaliÅ')],
-    ]
-
-    for test in tests:
-        test[0] = db.get_nick_id(test[2], create=True)
-        nick_id, slug, nick = test
-        registered = session.query(Nicknames) \
-                            .filter(Nicknames.canonical == nick) \
-                            .all()
-        assert len(registered) == 1
-        assert registered[0].slug == slug and registered[0].canonical == nick
-
-    # Check that each nick ended up with a different id
-    assert len(set(test[0] for test in tests)) == len(tests)
+    """Test get_nick_id does not create NickID by default."""
+    nick = Identifier('Exirel')
+    session = db.session()
+
+    # Attempt to get NickIDs: it is not created by default
+    with pytest.raises(ValueError):
+        db.get_nick_id(nick)
+
+    # Create the NickIDs
+    nick_id = db.get_nick_id(nick, create=True)
+
+    # Check that one and only one Nicknames exist with that ID
```

Comment:
I recall now the danger of asking you to write comments sometimes: the nemesi**S** can strike 😹

```suggestion
    # Attempt to get NickID: it is not created by default
    with pytest.raises(ValueError):
        db.get_nick_id(nick)

    # Create the NickID
    nick_id = db.get_nick_id(nick, create=True)

    # Check that one and only one Nickname exists with that ID
```

Reply:
The classes are named `NickIDs` and `Nicknames` tho. 😁

Reply:
The test isn't checking for instances of the classes, though. The classes represent database tables, but the test is checking for values. Maybe clearer in the `NickIDs` case to just replace it with `nick ID`, and lowercase `nickname`?

Reply:
Yes, as per our IRC discussion!